pilot/pkg/proxy/envoy/v1/watcher.go

```diff
@@ -116,7 +116,10 @@ func (w *watcher) Reload() {
 // retrieveAZ will only run once and then exit because AZ won't change over a proxy's lifecycle
 // it has to use a reload due to limitations with envoy (az has to be passed in as a flag)
 func (w *watcher) retrieveAZ(ctx context.Context, delay time.Duration, retries int) {
-	if !model.IsApplicationNodeType(w.role.Type) {
+	if w.config.GetControlPlaneAuthPolicy() == meshconfig.AuthenticationPolicy_MUTUAL_TLS {
+		log.Infof("Zone aware routing is not currently supported with mutual TLS enabled")
```

Comment:
Nit: Do we have a style guide that permits MutualTLS or Mutual-TLS? If not it would be great to have the log message in one of those styles.

Reply:
Would be great to have a small test case to capture this as well.

Reply:
Is it possible that if we see the Mutual_TLS flag we can skip calling this function altogether?

Reply:
We could but I feel this way it makes it more obvious to people who may expect it to work. It keeps it encapsulated in the function as well.

Reply:
Have added a test case and I have no idea about the style guide.